pkg/apiserver/certificate/selfsignedcert_provider.go

```diff
@@ -317,7 +317,8 @@ func (p *selfSignedCertProvider) saveCertKeyToSecret(secret *corev1.Secret, cert
 		if bytes.Equal(cert, secret.Data[corev1.TLSCertKey]) && bytes.Equal(key, secret.Data[corev1.TLSPrivateKeyKey]) {
 			return nil
 		}
-		secret.Type = corev1.SecretTypeTLS
+		// Do not update the existing Secret's type. Otherwise, the update would fail if it's not of type
```

Comment:
Should we fail the save action if the secret is not either tls nor opaque type?

Reply:
We don't actually care the type. Even though it doesn't make sense to use other types to store the secret, it can still work if user really set other types. And the point of the change is to be more robust, I think checking something we don't care is not helpful.